tools/common/StaticRegistrar.cs

```diff
@@ -2767,7 +2781,7 @@ void Specialize (AutoIndentStringBuilder sb)
 					if (token_ref == uint.MaxValue)
 						token_ref = CreateTokenReference (@class.Type, TokenType.TypeDef);
 					protocol_wrapper_map.Add (token_ref, new Tuple<ObjCType, uint> (@class, CreateTokenReference (@class.ProtocolWrapperType, TokenType.TypeDef)));
-					if (needs_protocol_map) {
```

Comment:
Seems like it would be simpler to set `needs_protocol_map` to true like this somewhere further up:

```csharp
if (TryGetAttribute (@class.Type, "Foundation", "XpcInterfaceAttribute", out var xpcAttr))
    needs_protocol_map = true;
```

That would make the next change below redundant as well.